exonum-java-binding/core/rust/src/storage/proof_map_index.rs

```diff
@@ -63,18 +88,34 @@ pub extern "system" fn Java_com_exonum_binding_core_storage_indices_ProofMapInde
     _: JClass,
     name: JString,
     view_handle: Handle,
-    // TODO: to be used in ECR-3765
-    _key_hashing: jboolean,
+    key_hashing: jboolean,
 ) -> Handle {
     let res = panic::catch_unwind(|| {
         let name = utils::convert_to_string(&env, name)?;
+        let key_is_hashed = key_hashing != JNI_FALSE;
         Ok(handle::to_handle(
             match handle::cast_handle::<View>(view_handle).get() {
                 ViewRef::Snapshot(snapshot) => {
-                    IndexType::SnapshotIndex(Index::from_access(snapshot, name.into()).unwrap())
+                    if key_is_hashed {
+                        IndexType::SnapshotIndex(Index::Hashed(
```

Comment:
Looks like IndexType::SnapshotIndex can be moved one level above, to remove some code duplication